### ISSUE TITLE
data uri to import other maps

### ISSUE THREAD

**Issue by zz-ha-dum** [reloZid/android-anuto#143]
map geometry, and enemy flows.

Share maps via email, jabber texting, or free hosted wiki.  No filesystem interface required

another app to make levels or webtool

**Comment by reloZid** [comment]
Sounds good, but the editor probably requires more effort than the game itself.

**Comment by LienRag** [comment]
Indeed, but couldn't you just explain the desired format and let people make it themselves if they have enough motivation?
You just need to include a way to install these new maps.

**Comment by cyphase** [comment]
@LienRag [Here's the map data for reference](https://github.com/reloZid/android-anuto/tree/master/app/src/main/res/raw). It shouldn't be too difficult to create a tool to make it easier to create maps (versus hand-typing JSON). How have you done it in the past @reloZid?

Once a map is created, maybe you could get it merged. Worst case you could build your own version of the app.

Open source +1.

**Comment by reloZid** [comment]
I always edited the JSON by hand and I guess the other contributors did too. I'm always open for new map proposal via pull request. But yes, this means that you need the full development environment to create maps.

**Comment by LienRag** [comment]
You mean this one for example? https://github.com/reloZid/android-anuto/blob/master/app/src/main/res/raw/map_cloverleaf.json 
Each (x,y) of name "basic" is a block?
I don't get negative coordinates for waypoints, though.

What would happen if height and width differed from (15,10)?
Bigger maps would be interesting if they are easy to navigate (I guess sliding only in one direction, else it gets very messy)...

**Comment by reloZid** [comment]
> Each (x,y) of name "basic" is a block?

Correct. It's called "basic" because one could implement special types of blocks.

> I don't get negative coordinates for waypoints, though.

Each path should start and end outside the visible area (otherwise you'll see how the enemies spawn or disappear). Therefore negative coordinates are needed.

> What would happen if height and width differed from (15,10)?

You can choose any width/height but for bigger maps the tiles will get smaller such that the map fits the screen which might become an issue on small screens. Currently there is no sliding implemented.
